File: PROJECT/pharma-reverse-chain-platform/backend/app/seed_data.py
SEED_ACTORS = [
    {"actor_id": "PHARMACY_001", "name": "CityCare Pharmacy", "role": "Pharmacy", "location": "New York, NY"},
    {"actor_id": "PHARMACY_002", "name": "GreenCross Pharmacy", "role": "Pharmacy", "location": "Austin, TX"},
    {"actor_id": "DISTRIBUTOR_001", "name": "MediTrans Logistics", "role": "Distributor", "location": "Chicago, IL"},
    {"actor_id": "DISTRIBUTOR_002", "name": "SwiftPharma Distributors", "role": "Distributor", "location": "Atlanta, GA"},
    {"actor_id": "MANUFACTURER_001", "name": "NovaGen Labs", "role": "Manufacturer", "location": "Boston, MA"},
    {"actor_id": "REGULATOR_001", "name": "National Drug Authority", "role": "Regulator", "location": "Washington, DC"},
]

# username / password / role / actor_id / display name
SEED_USERS = [
    {"username": "pharmacy1", "password": "pharmacy123", "role": "Pharmacy", "actor_id": "PHARMACY_001", "name": "CityCare Pharmacy"},
    {"username": "pharmacy2", "password": "pharmacy123", "role": "Pharmacy", "actor_id": "PHARMACY_002", "name": "GreenCross Pharmacy"},
    {"username": "distributor1", "password": "distributor123", "role": "Distributor", "actor_id": "DISTRIBUTOR_001", "name": "MediTrans Logistics"},
    {"username": "manufacturer1", "password": "manufacturer123", "role": "Manufacturer", "actor_id": "MANUFACTURER_001", "name": "NovaGen Labs"},
    {"username": "regulator1", "password": "regulator123", "role": "Regulator", "actor_id": "REGULATOR_001", "name": "National Drug Authority"},
]

SEED_BATCHES = [
    {"batch_number": "NV-AMX-1001", "medicine_name": "Amoxicillin 500mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2027-06-30", "quantity": 5000, "target": "ACTIVE"},
    {"batch_number": "NV-PAR-1002", "medicine_name": "Paracetamol 650mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2026-11-15", "quantity": 12000, "target": "ACTIVE"},
    {"batch_number": "NV-IBU-1003", "medicine_name": "Ibuprofen 400mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2025-12-01", "quantity": 8000, "target": "LOGGED_FOR_RETURN"},
    {"batch_number": "NV-MET-1004", "medicine_name": "Metformin 850mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2026-03-20", "quantity": 6000, "target": "PICKUP_SCHEDULED"},
    {"batch_number": "NV-OMZ-1005", "medicine_name": "Omeprazole 20mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2025-09-10", "quantity": 4500, "target": "RECEIVED_BY_DISTRIBUTOR"},
    {"batch_number": "NV-AZI-1006", "medicine_name": "Azithromycin 250mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2026-01-25", "quantity": 3000, "target": "DISPUTED"},
    {"batch_number": "NV-LOS-1007", "medicine_name": "Losartan 50mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2025-08-05", "quantity": 7200, "target": "WAITING_FOR_DESTRUCTION"},
    {"batch_number": "NV-CET-1008", "medicine_name": "Cetirizine 10mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2024-12-31", "quantity": 15000, "target": "DESTROYED"},
    {"batch_number": "NV-ATOR-1009", "medicine_name": "Atorvastatin 20mg", "manufacturer_name": "NovaGen Labs", "expiry_date": "2027-02-14", "quantity": 9000, "target": "ACTIVE"},
    {"batch_number": "NV-INS-1010", "medicine_name": "Insulin Glargine 100IU", "manufacturer_name": "NovaGen Labs", "expiry_date": "2025-10-30", "quantity": 1200, "target": "DESTROYED"},
]


def path_to(target: str) -> list[str]:
    return {
        "ACTIVE": [],
        "LOGGED_FOR_RETURN": ["LOGGED_FOR_RETURN"],
        "PICKUP_SCHEDULED": ["LOGGED_FOR_RETURN", "PICKUP_SCHEDULED"],
        "RECEIVED_BY_DISTRIBUTOR": ["LOGGED_FOR_RETURN", "PICKUP_SCHEDULED", "RECEIVED_BY_DISTRIBUTOR"],
        "DISPUTED": ["LOGGED_FOR_RETURN", "PICKUP_SCHEDULED", "RECEIVED_BY_DISTRIBUTOR", "DISPUTED"],
        "WAITING_FOR_DESTRUCTION": ["LOGGED_FOR_RETURN", "PICKUP_SCHEDULED", "RECEIVED_BY_DISTRIBUTOR", "WAITING_FOR_DESTRUCTION"],
        "DESTROYED": ["LOGGED_FOR_RETURN", "PICKUP_SCHEDULED", "RECEIVED_BY_DISTRIBUTOR", "WAITING_FOR_DESTRUCTION", "DESTROYED"],
    }.get(target, [])


ACTOR_FOR_STATE = {
    "LOGGED_FOR_RETURN": ("Pharmacy", "CityCare Pharmacy", "Return logged: expired / excess stock"),
    "PICKUP_SCHEDULED": ("Distributor", "MediTrans Logistics", "Pickup scheduled with pharmacy"),
    "RECEIVED_BY_DISTRIBUTOR": ("Distributor", "MediTrans Logistics", "Batch received at distributor warehouse"),
    "DISPUTED": ("Distributor", "MediTrans Logistics", "Quantity mismatch: counted vs declared"),
    "WAITING_FOR_DESTRUCTION": ("Manufacturer", "NovaGen Labs", "Accepted for certified destruction"),
    "DESTROYED": ("Manufacturer", "NovaGen Labs", "Destroyed under supervision. Certificate issued."),
}

File: PROJECT/pharma-reverse-chain-platform/src/lib/seed-data.ts
export const SEED_ACTORS = [
  { actorId: "PHARMACY_001", name: "CityCare Pharmacy", role: "Pharmacy", location: "New York, NY" },
  { actorId: "PHARMACY_002", name: "GreenCross Pharmacy", role: "Pharmacy", location: "Austin, TX" },
  { actorId: "DISTRIBUTOR_001", name: "MediTrans Logistics", role: "Distributor", location: "Chicago, IL" },
  { actorId: "DISTRIBUTOR_002", name: "SwiftPharma Distributors", role: "Distributor", location: "Atlanta, GA" },
  { actorId: "MANUFACTURER_001", name: "NovaGen Labs", role: "Manufacturer", location: "Boston, MA" },
  { actorId: "REGULATOR_001", name: "National Drug Authority", role: "Regulator", location: "Washington, DC" },
];

export type SeedBatch = {
  batchNumber: string;
  medicineName: string;
  manufacturerName: string;
  expiryDate: string;
  quantity: number;
  targetStatus: string;
};

export const SEED_BATCHES: SeedBatch[] = [
  {
    batchNumber: "NV-AMX-1001",
    medicineName: "Amoxicillin 500mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2027-06-30",
    quantity: 5000,
    targetStatus: "ACTIVE",
  },
  {
    batchNumber: "NV-PAR-1002",
    medicineName: "Paracetamol 650mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2026-11-15",
    quantity: 12000,
    targetStatus: "ACTIVE",
  },
  {
    batchNumber: "NV-IBU-1003",
    medicineName: "Ibuprofen 400mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2025-12-01",
    quantity: 8000,
    targetStatus: "LOGGED_FOR_RETURN",
  },
  {
    batchNumber: "NV-MET-1004",
    medicineName: "Metformin 850mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2026-03-20",
    quantity: 6000,
    targetStatus: "PICKUP_SCHEDULED",
  },
  {
    batchNumber: "NV-OMZ-1005",
    medicineName: "Omeprazole 20mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2025-09-10",
    quantity: 4500,
    targetStatus: "RECEIVED_BY_DISTRIBUTOR",
  },
  {
    batchNumber: "NV-AZI-1006",
    medicineName: "Azithromycin 250mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2026-01-25",
    quantity: 3000,
    targetStatus: "DISPUTED",
  },
  {
    batchNumber: "NV-LOS-1007",
    medicineName: "Losartan 50mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2025-08-05",
    quantity: 7200,
    targetStatus: "WAITING_FOR_DESTRUCTION",
  },
  {
    batchNumber: "NV-CET-1008",
    medicineName: "Cetirizine 10mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2024-12-31",
    quantity: 15000,
    targetStatus: "DESTROYED",
  },
  {
    batchNumber: "NV-ATOR-1009",
    medicineName: "Atorvastatin 20mg",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2027-02-14",
    quantity: 9000,
    targetStatus: "ACTIVE",
  },
  {
    batchNumber: "NV-INS-1010",
    medicineName: "Insulin Glargine 100IU",
    manufacturerName: "NovaGen Labs",
    expiryDate: "2025-10-30",
    quantity: 1200,
    targetStatus: "DESTROYED",
  },
];

// Walk the legal path from ACTIVE to the target so the audit history is realistic.
export function pathTo(target: string): string[] {
  switch (target) {
    case "ACTIVE":
      return [];
    case "LOGGED_FOR_RETURN":
      return ["LOGGED_FOR_RETURN"];
    case "PICKUP_SCHEDULED":
      return ["LOGGED_FOR_RETURN", "PICKUP_SCHEDULED"];
    case "RECEIVED_BY_DISTRIBUTOR":
      return ["LOGGED_FOR_RETURN", "PICKUP_SCHEDULED", "RECEIVED_BY_DISTRIBUTOR"];
    case "DISPUTED":
      return [
        "LOGGED_FOR_RETURN",
        "PICKUP_SCHEDULED",
        "RECEIVED_BY_DISTRIBUTOR",
        "DISPUTED",
      ];
    case "WAITING_FOR_DESTRUCTION":
      return [
        "LOGGED_FOR_RETURN",
        "PICKUP_SCHEDULED",
        "RECEIVED_BY_DISTRIBUTOR",
        "WAITING_FOR_DESTRUCTION",
      ];
    case "DESTROYED":
      return [
        "LOGGED_FOR_RETURN",
        "PICKUP_SCHEDULED",
        "RECEIVED_BY_DISTRIBUTOR",
        "WAITING_FOR_DESTRUCTION",
        "DESTROYED",
      ];
    default:
      return [];
  }
}

export function actorForState(state: string): { role: string; name: string; remarks: string } {
  switch (state) {
    case "LOGGED_FOR_RETURN":
      return {
        role: "Pharmacy",
        name: "CityCare Pharmacy",
        remarks: "Return logged: expired / excess stock",
      };
    case "PICKUP_SCHEDULED":
      return {
        role: "Distributor",
        name: "MediTrans Logistics",
        remarks: "Pickup scheduled with pharmacy",
      };
    case "RECEIVED_BY_DISTRIBUTOR":
      return {
        role: "Distributor",
        name: "MediTrans Logistics",
        remarks: "Batch received at distributor warehouse",
      };
    case "DISPUTED":
      return {
        role: "Distributor",
        name: "MediTrans Logistics",
        remarks: "Quantity mismatch: counted 2,850 vs declared 3,000",
      };
    case "WAITING_FOR_DESTRUCTION":
      return {
        role: "Manufacturer",
        name: "NovaGen Labs",
        remarks: "Accepted for certified destruction",
      };
    case "DESTROYED":
      return {
        role: "Manufacturer",
        name: "NovaGen Labs",
        remarks: "Destroyed under supervision. Certificate issued.",
      };
    default:
      return { role: "System", name: "System", remarks: "Batch created" };
  }
}
